spec/models/article_spec.rb

```diff
@@ -16,6 +16,22 @@
     it { should validate_uniqueness_of :title }
   end
 
+  describe ".published" do
+    it "returns the articles that have a published_at timestamp before the current time" do
+      published_article = create :article
+      article_published_today = create :article, published_at: 3.hours.ago
+      _unpublished_article = create :article, :unpublished
+      _article_to_be_published_today = create :article, published_at: 3.hours.from_now
+
+      expect(Article.published.ids).to eq(
+        [
+          published_article.id,
+          article_published_today.id,
+        ],
+      )
```

Comment:
```suggestion
      expect(Article.published).to match_array(
        [
          published_article,
          article_published_today,
        ],
      )
```
1. Swapped `eq` to `match_array` so we don't care about order (unless you _do_ care about order in which case the scope can be ordered) (also, I _think_ that's the right matcher ... jeez I haven't done rails in so long)
1. Removed the `ids` for the full query - less performant so feel free to ignore but also I'm not sure about the balance of readability in specs vs performance. And I imagine when querying a table with 4 records in it, it should be pretty fast. Unless the records are huge. In which case maybe the base factory should create smaller records. But also, feel free to ignore this and leave the ids in!